diff --git a/node_modules/draft-js-mention-plugin/lib/MentionSuggestions/index.js b/node_modules/draft-js-mention-plugin/lib/MentionSuggestions/index.js
index ffabd49..30952d4 100644
--- a/node_modules/draft-js-mention-plugin/lib/MentionSuggestions/index.js
+++ b/node_modules/draft-js-mention-plugin/lib/MentionSuggestions/index.js
@@ -284,6 +284,10 @@ var MentionSuggestions = exports.MentionSuggestions = function (_Component) {
         if (keyboardEvent.keyCode === 9) {
           _this.onTab(keyboardEvent);
         }
+        // space
+        if (keyboardEvent.keyCode === 32 && !_this.props.supportWhitespace) {
+          _this.onTab(keyboardEvent);
+        }
       };
 
       var descendant = 'mention-option-' + _this.key + '-' + _this.state.focusedOptionIndex;
diff --git a/node_modules/draft-js-mention-plugin/lib/defaultRegExp.js b/node_modules/draft-js-mention-plugin/lib/defaultRegExp.js
index 6f00958..1f4f919 100644
--- a/node_modules/draft-js-mention-plugin/lib/defaultRegExp.js
+++ b/node_modules/draft-js-mention-plugin/lib/defaultRegExp.js
@@ -3,7 +3,7 @@
 Object.defineProperty(exports, "__esModule", {
   value: true
 });
-exports.default = '[' + '\\w-' +
+exports.default = '[' + '\\w-\\\\\\.,\\[\\]\\{\\};\\:\'\\"\\/\\?\\>\\<\\|=\\+\\`\\!\\@\\#\\$\\%\\^\\&\\*\\(\\)\\~' +
 // Latin-1 Supplement (letters only) - https://en.wikipedia.org/wiki/List_of_Unicode_characters#Latin-1_Supplement
 '\xC0-\xD6' + '\xD8-\xF6' + '\xF8-\xFF' +
 // Latin Extended-A (without deprecated character) - https://en.wikipedia.org/wiki/List_of_Unicode_characters#Latin_Extended-A
diff --git a/node_modules/draft-js-mention-plugin/lib/index.js b/node_modules/draft-js-mention-plugin/lib/index.js
index b48c33c..ebb4ef4 100644
--- a/node_modules/draft-js-mention-plugin/lib/index.js
+++ b/node_modules/draft-js-mention-plugin/lib/index.js
@@ -183,7 +183,8 @@ exports.default = function () {
     entityMutability: entityMutability,
     positionSuggestions: positionSuggestions,
     mentionTrigger: mentionTrigger,
-    mentionPrefix: mentionPrefix
+    mentionPrefix: mentionPrefix,
+    supportWhitespace: supportWhitespace
   };
   var DecoratedMentionSuggestionsComponent = function DecoratedMentionSuggestionsComponent(props) {
     return _react2.default.createElement(MentionSuggestionsComponent, _extends({}, props, mentionSearchProps));
diff --git a/node_modules/draft-js-mention-plugin/lib/mentionSuggestionsStrategy.js b/node_modules/draft-js-mention-plugin/lib/mentionSuggestionsStrategy.js
index 03cba8c..767e618 100644
--- a/node_modules/draft-js-mention-plugin/lib/mentionSuggestionsStrategy.js
+++ b/node_modules/draft-js-mention-plugin/lib/mentionSuggestionsStrategy.js
@@ -38,7 +38,7 @@ var findWithRegex = function findWithRegex(regex, contentBlock, callback) {
 
 exports.default = function (trigger, supportWhiteSpace, regExp) {
   //eslint-disable-line
-  var MENTION_REGEX = supportWhiteSpace ? new RegExp((0, _escapeRegExp2.default)(trigger) + '(' + regExp + '|\\s){0,}', 'g') : new RegExp('(\\s|^)' + (0, _escapeRegExp2.default)(trigger) + regExp, 'g');
+  var MENTION_REGEX = supportWhiteSpace ? new RegExp('(\\s|^)' + (0, _escapeRegExp2.default)(trigger) + '(' + regExp + '|\\s){0,}', 'g') : new RegExp('(\\s|^)' + (0, _escapeRegExp2.default)(trigger) + regExp, 'g');
 
   return function (contentBlock, callback) {
     findWithRegex(MENTION_REGEX, contentBlock, callback);
