diff --git a/node_modules/draft-js-mention-plugin/lib/mentionSuggestionsStrategy.js b/node_modules/draft-js-mention-plugin/lib/mentionSuggestionsStrategy.js
index 03cba8c..767e618 100644
--- a/node_modules/draft-js-mention-plugin/lib/mentionSuggestionsStrategy.js
+++ b/node_modules/draft-js-mention-plugin/lib/mentionSuggestionsStrategy.js
@@ -38,7 +38,7 @@ var findWithRegex = function findWithRegex(regex, contentBlock, callback) {
 
 exports.default = function (trigger, supportWhiteSpace, regExp) {
   //eslint-disable-line
-  var MENTION_REGEX = supportWhiteSpace ? new RegExp((0, _escapeRegExp2.default)(trigger) + '(' + regExp + '|\\s){0,}', 'g') : new RegExp('(\\s|^)' + (0, _escapeRegExp2.default)(trigger) + regExp, 'g');
+  var MENTION_REGEX = supportWhiteSpace ? new RegExp('(\\s|^)' + (0, _escapeRegExp2.default)(trigger) + '(' + regExp + '|\\s){0,}', 'g') : new RegExp('(\\s|^)' + (0, _escapeRegExp2.default)(trigger) + regExp, 'g');
 
   return function (contentBlock, callback) {
     findWithRegex(MENTION_REGEX, contentBlock, callback);
